sirepo/pkcli/job_cmd.py

```diff
@@ -217,6 +218,8 @@ def _change_message_too_big(msg):
     if type(msg) != bytes:
         msg = pkjson.dump_bytes(msg)
     if len(msg + b'\n') > job.cfg.max_message_bytes:
+        # TODO (gurhar1133): Do we need to change things so that we don't have to send COMPLETED for
+        # this error case?
         msg = pkjson.dump_bytes(PKDict(
```

Comment:
Formatting:

```py
msg = pkjson.dump_bytes(
    PKDict(
    [snip],
    ),
)
```

Avoid multiple calls on a single line when broken out. I think it is acceptable to

```py
msg = pkjson.dump_bytes(PKDict(data=job.COMPLETED, error='Data was too large'))
```

I perhaps set a bad example with resetting a parameter, which was discussed with @e-carlin. Here, `msg` goes through several transformation so I would code this as:

```py
def _validate_and_jsonl(msg):
    m = pkjson.dump_bytes(msg)
    if len(m) >=  job.cfg.max_message_bytes:
        m = pkjson.dump_bytes(PKDict(data=job.COMPLETED, error='Response is too large to send'))
    return m + b'\n'
```

I don't think `_change_message_too_big` is right, because it implies it is going to change. Rather validate implies it is conditional. The conversion to jsonl should happen here as well. This simplifies the code in the caller and documents that it is being converted to jsonl. There isn't a need for the conditional on the bytes, because it always comes in as a PKDict. If it is exactly max_message_bytes, the `>=` fixes that, which is an implicit +1 which I don't like so you could add a -1 if you feel it is clearer.

`b` is unnecessary afaict.

Reply:
b is for the case that we call _do_download_data_file
https://github.com/radiasoft/sirepo/blob/f922e02c7fe98a7d05610e50c322b019360ec243/sirepo/pkcli/job_cmd.py#L150-L155

But maybe something like:
```python
c = _validate_and_jsonl(c)
if 'error' in c:
    return c
```
would be better. I may be more involved with c is in bytes. I'll work on this.

Reply:
I looked for this, and somehow I didn't see it, sorry.

I think two functions are warranted: _validate_msg and _validate_msg_and_json.

Reply:
Is two functions better than
```python
c = _validate_and_jsonl(c)
    if 'error' in pkjson.load_any(c):
        return pkjson.load_any(c)
```
for _do_download_data_file?